assignment/impl/src/java/org/sakaiproject/assignment/impl/EmailUtil.java

```diff
@@ -0,0 +1,350 @@
+    @Setter private DeveloperHelperService developerHelperService;
+    @Setter private EntityManager entityManager;
+    @Setter private FormattedText formattedText;
+    @Setter private ResourceLoader resourceLoader;
+    @Setter private ServerConfigurationService serverConfigurationService;
+    @Setter private SiteService siteService;
+    @Setter private UserDirectoryService userDirectoryService;
+
+    public List<String> getHeaders(String receiverEmail, String submissionOrReleaseGrade) {
+        List<String> rv = new ArrayList<>();
+
+        rv.add("MIME-Version: 1.0");
+        rv.add("Content-Type: multipart/alternative; boundary=\"" + MULTIPART_BOUNDARY + "\"");
+        // set the subject
+        rv.add(getSubject(submissionOrReleaseGrade));
+
+        // from
+        rv.add(getFrom());
+
+        // to
+        if (StringUtils.isNotBlank(receiverEmail)) {
+            rv.add("To: " + receiverEmail);
+        }
+
+        return rv;
+    }
+
+    public String getSubject(String submissionOrReleaseGrade) {
+        String subject;
+        if ("submission".equals(submissionOrReleaseGrade)) {
+            subject = resourceLoader.getString("noti.subject.content");
+        } else if ("releasegrade".equals(submissionOrReleaseGrade)) {
+            subject = resourceLoader.getString("noti.releasegrade.subject.content");
+        } else {
+            subject = resourceLoader.getString("noti.releaseresubmission.subject.content");
+        }
+        return "Subject: " + subject;
+    }
+
+    public String getFrom() {
+        return "From: " + "\"" + serverConfigurationService.getString("ui.service", "Sakai") + "\" <" + serverConfigurationService.getString("setup.request", "no-reply@" + serverConfigurationService.getServerName()) + ">";
+    }
+
+    public String getNotificationMessage(AssignmentSubmission s, String submissionOrReleaseGrade) {
+        StringBuilder message = new StringBuilder();
+        message.append(MIME_ADVISORY);
+        message.append(BOUNDARY_LINE);
+        message.append(plainTextHeaders());
+        message.append(plainTextContent(s, submissionOrReleaseGrade));
+        message.append(formattedText.convertFormattedTextToPlaintext(htmlContentAttachments(s)));
+        message.append(BOUNDARY_LINE);
+        message.append(htmlHeaders());
+        message.append(htmlPreamble(submissionOrReleaseGrade));
+        if ("submission".equals(submissionOrReleaseGrade))
+            message.append(htmlContent(s));
+        else if ("releasegrade".equals(submissionOrReleaseGrade))
+            message.append(htmlContentReleaseGrade(s));
+        else
+            message.append(htmlContentReleaseResubmission(s));
+        message.append(htmlContentAttachments(s));
+        message.append(htmlEnd());
+        message.append(TERMINATION_LINE);
+        return message.toString();
+    }
+
+    public String plainTextHeaders() {
+        return "Content-Type: text/plain\n\n";
```

Comment:
are you sure you want double line break?

Reply:
Yeah thats how it was previously, but I agree not sure why it was like that.

Reply:
yeah, i remember seeing issues with PHP and multiple line breaks telling the email client that the body was starting .... so if the body is indeed next, then this seems fine

https://stackoverflow.com/questions/30887610/error-with-php-mail-multiple-or-malformed-newlines-found-in-additional-header

Reply:
seems like the contents follow so this is indeed correct